src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -63,7 +63,7 @@ public static int IndexOfAny(ref byte searchSpace, int searchSpaceLength, ref by
             int index = -1;
             for (int i = 0; i < valueLength; i++)
             {
-                var tempIndex = IndexOf(ref searchSpace, Unsafe.Add(ref value, i), searchSpaceLength);
+                var tempIndex = IndexOf(ref searchSpace, Unsafe.Add(ref value, i), index < 0 ? searchSpaceLength : index);
```

Comment:
```cs
if (index < 0) index = searchSpaceLength
var tempIndex = IndexOf(ref searchSpace, Unsafe.Add(ref value, i), index);
```
should give a better code from the JIT

Reply:
Can you please add a comment explaining why passing index as the length works here when index != -1.
IndexOf takes in a length of the search space, but index would point to the location of one of the values being searched (not the length). This works because we are looking for the first index of any of the search values, correct (since we don't care if we find the search value after the index of a previously found value)?

Would something like this work, where you don't need the `index` local variable?
```C#
for (int i = 0; i < valueLength && index != 0; i++)
{
    var tempIndex = IndexOf(ref searchSpace, Unsafe.Add(ref value, i), searchSpaceLength);
    if ((uint)tempIndex < (uint)searchSpaceLength)
        searchSpaceLength = tempIndex;
}
```

Reply:
Yes, you are right, passing index if it's not equal -1 allows to search values only before founded solution. I also come to idea to use only searchSpaceLength. Does JIT produce code which allows in-lining methods with reassigning arguments (I've heard that RuyJIT in .NET Core 1.0 didn't support in-lining such methods)?